### ISSUE TITLE
Raw return type seems off

### ISSUE THREAD

**Issue by kevinthecity** [knex/knex#3341]
# Environment

Knex version: 0.18.3
Database + version: postgres
OS: macOS 10.14.4
VSCode: 1.36.0
TS: 3.5.3

# Bug

When making a raw query, the results returns `any[]`, however the actual object seems to be a results object.

Previously I was returning `results.rows`, however now the bindings now say that the return type of raw(...) is `any[]`, however, printing indicated that it is a results object. I just upgraded from `"knex": "^0.16.3",` and saw this behavior.

It _could_ be related to this change - https://github.com/tgriesser/knex/commit/9c6ab963f6f6af135ef72b6d5fdfb14b6f3707c6#diff-88bec6beae84369c0376b56b3bb88fe1R268

```js
const queryDb = () => {
  const results = await db.query.raw(rawQueryBuilder.join(" "), bindings);
  console.log(results);

  // Use of ts-ignore allows it to work as it did before
  // @ts-ignore
  return results.rows;
}

// Results of log
Result {
  command: 'SELECT',
  rowCount: 8,
  oid: null,
  rows: [...],
  fields: [...],
  parsers: [...],
  _types: TypeOverrides {
    _types: {
      getTypeParser: [Function: getTypeParser],
      setTypeParser: [Function: setTypeParser],
      arrayParser: [Object]
    },
    text: {},
    binary: {}
  },
  RowCtor: null,
  rowAsArray: false
}
```

Screenshot of IDE indicating the return type

![Screen Shot 2019-07-10 at 5 32 01 PM](https://user-images.githubusercontent.com/1014106/61006578-ac8e1700-a338-11e9-919a-67eb080c69b3.png)

![Screen Shot 2019-07-10 at 11 27 38 AM](https://user-images.githubusercontent.com/1014106/60982267-c8c49080-a305-11e9-9f85-515089c5165f.png)

# Expect

I would expect the return type of a `raw` query to return something like this (or at least

```
interface RawQueryResult<T>{
  command: string;
  rowCount: number;
  rows: T[]; // Or even just rows: any[];
  ...
}

```

**Comment by kevinthecity** [comment]
@lorefnon tagging you on this as per the template! 🙏

**Comment by elhigu** [comment]
Yep... raw return type depends on driver. Usually it is not an array.

**Comment by lorefnon** [comment]
Yes this is an issue with the typings. I am travelling at the moment but I'll submit a pr for this shortly.

**Comment by kibertoad** [comment]
Released in 0.19.1
